samples/cpp/cloning_gui.cpp

```diff
@@ -453,7 +453,7 @@ int main()
 
     for(;;)
     {
-        char key = (char) waitKey(0);
+        int key = waitKey(0) & 0xFF;
 
         if(key == 'd' && flag3 == 0)
```

Comment:
'd' literal constant is treated as char that could be internally represented as signed char(that is right for most of systems I know) or unsigned char. It doesn't make a difference for 'd' const which turn to value below 127 and stay the same anyway. But if user tries to compare against a symbol from upper half of ascii table it could became rather positive value greater than 127 or negative value depending on internal char representation.